lib/thrift/generator/struct_binary_protocol.ex

```diff
@@ -769,30 +769,55 @@ defmodule Thrift.Generator.StructBinaryProtocol do
     end
   end
 
-  defp field_serializer(%Field{name: name, type: :bool, id: id}, _file_group) do
-    var = Macro.var(name, nil)
+  defp field_serializer(%Field{name: name, type: :bool, id: id, required: false}, _file_group) do
     quote do
-      case unquote(var) do
+      case unquote(Macro.var(name, nil)) do
         nil ->
           <<>>
         false ->
           <<unquote(@bool), unquote(id) :: size(16), 0>>
+        true ->
+          <<unquote(@bool), unquote(id) :: size(16), 1>>
         _ ->
+          raise Thrift.MissingFieldException,
+                unquote("Optional boolean field #{inspect name} must be true, false, or nil")
```

Comment:
It would be helpful to also include the struct/module name in addition to the field name, if possible. Below as well.